File: XamarinXamlPrev/ViewModel/DataModel.fs
namespace XamarinXamlPrev
open System
open System.Collections.Generic
open System.Linq
open System.Text
open Xamarin.Forms
open Moonmile.XFormsProvider
open System.IO
open System.Reflection

type DataMode() =
    inherit ViewModelBase()

    let mutable appName = "xaml preview"
    let mutable version = "alpha"

    member this.AppName
        with get() = appName
    member this.Version
        with get() = version
        and set(v) = version <- v
    member this.Updated 
        with get() = DateTime.Now.ToString()

type ViewModelNextPage() =
    inherit ViewModelBase()

    let mutable updated  = DateTime.Now

    member this.Updated 
        with get() = updated.ToString()
        and set(value) = 
            if updated.ToString() <> value then
                updated <- DateTime.Parse( value )
                base.OnPropertyChanged("Updated")

    member this.UpdatedCommand 
        with get() =
            let cmd = 
                new DelegateCommand(
                    (fun (o) -> true ),
                    (fun (o) -> this.Updated <- DateTime.Now.ToString()))
            cmd

File: CSharp/XamarinXamlPreviewCS/XamarinXamlPreviewCS/XamarinXamlPreviewCS/ViewModel/DataModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Input;
using XamarinXamlPreviewCS.Libs;

namespace XamarinXamlPreviewCS.ViewModel
{
    public class DataModel : ViewModelBase
    {
        public string AppName { get; internal set; }
        public string Version { get; internal set; }
        public string Updated { get { return DateTime.Now.ToString(); } }

        public DataModel()
        {
            this.AppName = "Xaml preview";
            this.Version = "Alpha version";
        }
    }

    public class ViewModelNextPage : ViewModelBase
    {
        private DateTime updated = DateTime.Now;

        public string Updated
        {
            get { return updated.ToString(); }
            set
            {
                if (updated.ToString() != value)
                {
                    updated = DateTime.Parse(value);
                    base.OnPropertyChanged("Updated");
                }
            }
        }
        public ICommand UpdatedCommand
        {
            get
            {
                var cmd = new DelegateCommand(
                    (s) => { return true; },
                    (s) => { this.Updated = DateTime.Now.ToString(); });
                return cmd;
            }
        }
    }
}
